### ISSUE TITLE
twi.h not found

### ISSUE THREAD

**Issue by andrewraharjo** [collin80/GVRET#18]
Collin, 

I have tried to run the GVRET build with due_wire and all dependencies in placed on the Content/Library. When I've tried to compile, it has given me this error

```
Arduino: 1.6.8 Hourly Build 2016/02/19 11:12 (Mac OS X), Board: "Arduino/Genuino Uno"

In file included from 
/Applications/Arduino.app/Contents/Java/libraries/due_wire/due_wire.h:25:25: fatal error: include/twi.h: No such file or directory
 #include <include/twi.h>
                         ^
compilation terminated.
exit status 1
Error compiling.

This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.
```

I have downloaded twi.h and required dependencies but it seems it should be inside the Arduino Application contents. Any clue ?

Thanks,
Andrew

**Comment by collin80** [comment]
Luckily, this is an easy to solve and very common problem. You see, you've
got the Arduino Uno selected as your board. That's not right. You need to
select the Arduino Due as your board. If you don't have that option then
you'll need to use the board manager to download the Arduino Due support
files.

**Comment by andrewraharjo** [comment]
@collin80 Good catch Collin. I have tried to run it again and changed the board target to Arduino Due (Native USB Port). It passed the build but I still got the Warning messages:

`WARNING: Category '' in library SPI is not valid. Setting to 'Uncategorized'
WARNING: Category '' in library Wire is not valid. Setting to 'Uncategorized'`

`Warning: platform.txt from core 'Arduino ARM (32-bits) Boards' contains deprecated recipe.c.combine.pattern="{compiler.path}{compiler.c.elf.cmd}" {compiler.c.elf.flags} -mcpu={build.mcu} "-T{build.variant.path}/{build.ldscript}" "-Wl,-Map,{build.path}/{build.project_name}.map" {compiler.c.elf.extra_flags} -o "{build.path}/{build.project_name}.elf" "-L{build.path}" -mthumb -Wl,--cref -Wl,--check-sections -Wl,--gc-sections -Wl,--entry=Reset_Handler -Wl,--unresolved-symbols=report-all -Wl,--warn-.....`

`Warning: platform.txt from core 'Arduino ARM (32-bits) Boards' contains deprecated recipe.ar.pattern="{compiler.path}{compiler.ar.cmd}" {compiler.ar.flags} {compiler.ar.extra_flags} ......
`
Is it normal ? I am running build 1.6.9

**Comment by andrewraharjo** [comment]
@collin80 So I have fixed the issue:
1. Remove ~/Library/Arduino15/ for all the SAMD packages
2. Update the Arduino IDE - Updated to hourly build 1.6.13
3. Copy all the required libraries for GVRET
4. Install SAMD for ARM32
5. Choose the Due Board - Native USB and I got it work

Is there any more details documentation how to use GVRET instead use http://store.evtv.me/proddetail.php?prod=EVTVDue2 ? Or the CANSkeleton should do the job ?

Thanks Collin

**Comment by collin80** [comment]
There probably should be some documentation but really there isn't. I think
that the serial console is pretty easy to use, there are some single letter
commands and some longer commands that take the form "something=value" such
as "can0en=1" to enable CAN0. These work if you have line endings of some
sort enabled (CR, LF, CRLF). Once you've set the speed of a CAN bus and
enabled it you should be able to send and receive with the board even from
the serial console. But, it tends to be simpler to use SavvyCAN which knows
how to talk to GVRET and has a GUI. Or, if you use linux there is
experimental socketcan support. A GVRET board will present itself as a
serial socketcan device (lawicel compatible). Then you can use whatever
linux tools you want.

**Comment by andrewraharjo** [comment]
@collin80 Cool, thanks for your patience answering my issue. I'm trying to understand how to utilize the tools for my current development. I tried to run SerialConsole.cpp to make sure everything works. It seems I'm in the right path to look over the data (nothing connected yet).

I have downloaded the SavvyCAN and ran it under OSX. I think I should spend more time to understand how to use it. Any idea where to start ? Instead the docs from your github ?
